Queries as signals - runtime follow-up (#54103)
angular/angular: adev/src/content/reference/errors/NG0951.md

@name Child query result is required but no value is available.
@category runtime
@shortDescription Required child query result was accessed before query results were calculated or query has no matches.

@description

> Required child query (`@ContentChild` or `@ViewChild`) result was accessed before query results were calculated or query has no matches.

Reviewer: It would be good to link to the docs for queries here

https://angular.dev/guide/components/queries

> This can happen in two distinct situations:

Reviewer 2: ```suggestion
This can happen in two situations:
```

* query results were accessed before a given query could collect results;
* a query was executed but didn't match any nodes and has no results as a consequence.

> Different types of queries calculate their results at different points in time:

Reviewer: ```suggestion
Content queries and view queries each calculate their results at different points in time:
```

Author: +1

* `@ContentChild` results are available after a _host_ view (template where a directive declaring a query is used) is created;
* `@ViewChild` results are available after a template of a component declaring a query is created.

> Accessing query results before those points in time will result in the error discussed here. Most notably, query results are _never_ available in a constructor of a directive declaring a query.

Reviewer: ```suggestion
Accessing query results before they're available results in the error described on this page. Most notably, query results are _never_ available in a constructor of the component or directive declaring a query.
```

> ## Fixing the error

Reviewer: I'd also link to the docs for lifecycle here 

https://angular.dev/guide/components/lifecycle

`@ContentChild` query results can be accessed in the `AfterContentChecked` lifecycle hook, or later.
`@ViewChild` query results can be accessed in the `AfterViewChecked` lifecycle hook, or later.

> Make sure that a required query matches at least one node and has results at all (you can verify it accessing query results in the lifecycle hooks listed above).

Reviewer: ```suggestion
Make sure that a required query matches at least one node and has results at all. You can verify this by accessing query results in the lifecycle hooks listed above.
```

Author: +1
